pkg/kn/commands/service/export.go

```diff
@@ -0,0 +1,233 @@
+// Copyright © 2020 The Knative Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package service
+
+import (
+	"errors"
+
+	"sort"
+	"strconv"
+
+	"github.com/spf13/cobra"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+
+	"knative.dev/client/pkg/kn/commands"
+	clientservingv1 "knative.dev/client/pkg/serving/v1"
+	"knative.dev/serving/pkg/apis/serving"
+	servingv1 "knative.dev/serving/pkg/apis/serving/v1"
+)
+
+// NewServiceExportCommand returns a new command for exporting a service.
+func NewServiceExportCommand(p *commands.KnParams) *cobra.Command {
+
+	// For machine readable output
+	machineReadablePrintFlags := genericclioptions.NewPrintFlags("")
+
+	command := &cobra.Command{
+		Use:   "export NAME",
+		Short: "export a service",
+		Example: `
+  # Export a service in yaml format
+  kn service export foo -n bar -o yaml
+  # Export a service in json format
+  kn service export foo -n bar -o json`,
+		RunE: func(cmd *cobra.Command, args []string) error {
```

Comment:
Can we split this long (100+ lines) `RunE` function into smaller more manageable private functions with clear names? That would make this much easier to read, review, debug, and test. Thx.

Reply:
Thanks for your comments @maximilien . Have tried my best to keep the ```RunE``` func short (50 lines :-)) . Since the flows (export service with and without history) return differnt objects I had to call that from RunE func.